flink-runtime/src/main/java/org/apache/flink/runtime/execution/librarycache/BlobLibraryCacheManager.java

```diff
@@ -124,7 +129,13 @@ public void registerTask(JobID jobId, ExecutionAttemptID task, Collection<BlobKe
 					ExceptionUtils.tryRethrowIOException(t);
 					throw new IOException("Library cache could not register the user code libraries.", t);
 				}
-				
+
+				// add classpaths
+				URL[] cpUrls = requiredClasspaths.toArray(new URL[requiredClasspaths.size()]);
```

Comment:
Any reason you're creating an array here first instead of simply iterating over `requiredClassPaths`?